int32 m1_speed
int32 m2_speed
int32 m1_position
int32 m2_position

